libs/community/langchain_community/chat_message_histories/sql.py

```diff
@@ -80,36 +108,98 @@ def get_sql_model_class(self) -> Any:
         return self.model_class
 
 
+DBConnection = Union[AsyncEngine, Engine, str]
+
+_warned_once_already = False
+
+
 class SQLChatMessageHistory(BaseChatMessageHistory):
     """Chat message history stored in an SQL database."""
 
+    @property
+    @deprecated("0.2.2", removal="0.3.0", alternative="session_maker")
+    def Session(self) -> Union[scoped_session, async_sessionmaker]:
+        return self.session_maker
+
     def __init__(
         self,
         session_id: str,
-        connection_string: str,
+        connection_string: Optional[str] = None,
         table_name: str = "message_store",
         session_id_field_name: str = "session_id",
         custom_message_converter: Optional[BaseMessageConverter] = None,
+        connection: Union[None, DBConnection] = None,
```

Comment:
(unrelated to this PR)

---

@pprados this pattern could be useful if you needed a transaction (e.g., as was the case in the indexing API if I recall correctly)

Allows passing a session maker as part of the __init__ instead of initializing and then mutating a class attribute (the latter pattern is more likely to break in threaded scenarios)